torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -1241,6 +1418,73 @@ bool isNodeParsible(const Node* node) {
   return IrParser::canParseNode(node);
 }
 
+bool insertProfileIValue(ProfilingRecord* pr, Node* node, size_t offset) {
+  // is skip constant necessary?
+  if (node->input(offset)->node()->kind() == prim::Constant) {
+    return false;
+  }
+
+  static auto reduction_operator_schema =
+      getOperatorForLiteral(
+          "aten::sum.dim_IntList(Tensor self, int[1] dim, bool keepdim=False, *, int? dtype=None) -> (Tensor)")
+          ->schema();
+  if (node->matches(reduction_operator_schema)) {
+    switch (offset) {
+      // argument 1: reduction axes;
+      case 1:
+        profileIntList(pr, node, offset);
+        break;
+      // argument 2: keepdim;
+      case 2:
+        profileBool(pr, node, offset);
+        break;
+      default:
+        return false;
+    }
+    return true;
+  }
+
+  static auto sum_to_size_schema =
+      getOperatorForLiteral(
+          "aten::sum_to_size(Tensor self, int[] size) -> Tensor")
+          ->schema();
+  static auto grad_sum_to_size_schema =
+      getOperatorForLiteral(
+          "aten::_grad_sum_to_size(Tensor(a) self, int[]? size) -> Tensor(a)")
+          ->schema();
+  if (node->matches(sum_to_size_schema) ||
+      node->matches(grad_sum_to_size_schema)) {
+    switch (offset) {
+      // argument 1: reduction sizes;
+      case 1:
+        // TODO(profile_size): double check optional[size]?
+        profileSize(pr, node, offset);
+        break;
+      default:
+        return false;
+    }
+    return true;
+  }
+
+  return false;
+}
+
+void insertProfileNodesForCUDAFuser_(Block* block, ProfilingRecord* pr) {
+  for (const auto& n : block->nodes()) {
+    for (size_t offset = 0; offset < n->inputs().size(); offset++) {
+      insertProfileIValue(pr, n, offset);
+    }
+
+    for (auto ib : n->blocks()) {
```

Comment:
Are you sure want to copy the block?

Reply:
`blocks()` returns an `at::ArrayRef<Block*>` so it's a trivial copy.